axiom/test/test_reference.py

```diff
@@ -187,18 +187,35 @@ def test_dummyItemReferenceUpgraded(self):
         """
         store = Store()
         referent = SimpleReferent(store=store)
-        oldReferee = nonUpgradedItem(store=store)
+        oldReferee = nonUpgradedItem2(store=store)
         referent.ref = oldReferee
         # Manually run the upgrader on this specific legacy item. This is the
         # same as if the SimpleReferent item had been created in an upgrader
         # for UpgradedItem, except that we can keep a strong reference to
         # oldReferee to ensure it is not garbage collected (this would
         # otherwise happen nondeterministically on platforms like PyPy).
-        newReferee = item1to2(oldReferee)
+        newReferee = item2to3(oldReferee)
         self.assertIsInstance(newReferee, UpgradedItem)
         self.assertIdentical(referent.ref, newReferee)
 
 
+    def test_dummyItemReferenceInUpgrade(self):
+        """
+        Setting the value of a reference attribute to a legacy item during an
+        upgrade does not lose the value.
```

Comment:
Minor nitpick - inasmuch as it's possible, test docstrings should be phrased positively.  So something like "preserves the value onto the upgraded item".